### PR TITLE
Prepare 1.7.x

### PR DESCRIPTION
**Pull request by Shubhranshu153**
Updating dependencies for the release

### CODE REVIEW

**Comment by Shubhranshu153** [comment]
@AkihiroSuda  This test `TestRunContainerWithMACAddress` seems to not docker compat after CNI plugins upgrade
Is the recommendation is to upgrade docker and port the fixes made in main to this branch?

**Comment by AkihiroSuda** [comment]
> @AkihiroSuda This test `TestRunContainerWithMACAddress` seems to not docker compat after CNI plugins upgrade Is the recommendation is to upgrade docker and port the fixes made in main to this branch?

I think you can rollback the Docker version (v25? v24?) on the CI and call it a day

Like this:
- https://github.com/containerd/nerdctl/pull/3089

File: Dockerfile [hunk 1]
```diff
@@ -19,8 +19,8 @@
 
 # Basic deps
 ARG CONTAINERD_VERSION=v1.7.16
-ARG RUNC_VERSION=v1.1.12
-ARG CNI_PLUGINS_VERSION=v1.4.1
+ARG RUNC_VERSION=v1.1.14
+ARG CNI_PLUGINS_VERSION=v1.5.1
 
 # Extra deps: Build
 ARG BUILDKIT_VERSION=v0.12.5
```
Comment (AkihiroSuda) [inline Dockerfile:26]: Probably we should update `CONTAINERD_VERSION`, `BUILDKIT_VERSION`, `ROOTLESSKIT_VERSION`, etc.

Comment (Shubhranshu153) [reply Dockerfile:26]: sure i can update those.
Probably also the test.yaml with the new versions.

Comment (Shubhranshu153) [reply Dockerfile:26]: For the ROOTLESSKIT does the version update be limited to minor version or we also update major?

Comment (Shubhranshu153) [reply Dockerfile:26]: nvm its a minor version upgrade will include it.

**Comment by Shubhranshu153** [comment]
@AkihiroSuda 
For test that pass i see errors:
```
=== Failed
=== FAIL: cmd/nerdctl TestLoginWithPlainHttp/{regHost:127.0.0.1_regPort:5000_useRegPort:true_username:admin_password:validTestPassword_shouldSuccess:true_registry:0xc00087c7e0_shouldUseInSecure:true} (0.04s)
    login_linux_test.go:172: Good password
    testing.go:1577: test executed panic(nil) or runtime.Goexit: subtest may have called FailNow on a parent test

=== FAIL: cmd/nerdctl TestLoginWithPlainHttp (7.07s)
    testregistry_linux.go:76: hostIP="10.4.0.1", listenIP="0.0.0.0", listenPort=5000, authPort=5001
    testregistry_linux.go:152: Writing "/tmp/TestLoginWithPlainHttp1769386958/004/certs.d3822826499/10.4.0.1:5000/hosts.toml": "\nserver = \"[https://10.4.0.1:5000\](https://10.4.0.1:5000/)"\n[host.\"[https://10.4.0.1:5000\](https://10.4.0.1:5000/)"]\n  ca = \"/tmp/TestLoginWithPlainHttp1769386958/001/ca2498508156/ca.cert\"\n\t\t"
    testregistry_linux.go:76: hostIP="10.4.0.1", listenIP="0.0.0.0", listenPort=80, authPort=5002
    testregistry_linux.go:152: Writing "/tmp/TestLoginWithPlainHttp1769386958/008/certs.d3819743510/10.4.0.1:80/hosts.toml": "\nserver = \"[https://10.4.0.1:80\](https://10.4.0.1:80/)"\n[host.\"[https://10.4.0.1:80\](https://10.4.0.1:80/)"]\n  ca = \"/tmp/TestLoginWithPlainHttp1769386958/005/ca1974659761/ca.cert\"\n\t\t"
    login_linux_test.go:185: assertion failed: res.ExitCode is not exitCode: time="2024-09-08T21:50:38Z" level=warning msg="WARNING! Using --password via the CLI is insecure. Use --password-stdin."
        time="2024-09-08T21:50:38Z" level=warning msg="skipping verifying HTTPS certs for \"127.0.0.1:5000\""
        time="2024-09-08T21:50:38Z" level=debug msg="Found hosts dir \"/tmp/TestLoginWithPlainHttp1769386958/004/certs.d3822826499\""
        time="2024-09-08T21:50:38Z" level=debug msg="len(regHosts)=1"
        time="2024-09-08T21:50:38Z" level=debug msg="no scope specified for token auth challenge" host="127.0.0.1:5000"
        time="2024-09-08T21:50:38Z" level=error msg="failed to call tryLoginWithRegHost" error="failed to call rh.Authorizer.Authorize: failed to fetch oauth token: Post \"[https://10.4.0.1:5001/auth\](https://10.4.0.1:5001/auth/)": tls: failed to verify certificate: x509: certificate signed by unknown authority" i=0
        time="2024-09-08T21:50:38Z" level=fatal msg="failed to call rh.Authorizer.Authorize: failed to fetch oauth token: Post \"[https://10.4.0.1:5001/auth\](https://10.4.0.1:5001/auth/)": tls: failed to verify certificate: x509: certificate signed by unknown authority"
        

DONE 2 runs, 475 tests, 8 skipped, 2 failures in 858.479s
```
is this expected?

The linter is failing due to caching it seems and there is some flakiness that is failing one of the matrix. Can we rerun those ones.

Thanks

**Comment by AkihiroSuda** [comment]
> test

Restarted. Seems green.

> linter

Probably golangci-lint has to be updated
https://github.com/containerd/nerdctl/blob/845e989f69d25b420ae325fedc8e70186243fd93/.github/workflows/test.yml#L48-L52

File: go.mod [hunk 2]
```diff
@@ -64,7 +65,12 @@ require (
 	gotest.tools/v3 v3.5.1
 )
 
-require github.com/moby/sys/user v0.1.0 // indirect
+require (
+	github.com/containerd/errdefs v0.1.0 // indirect
+	github.com/containerd/platforms v0.2.1 // indirect
+	github.com/moby/sys/user v0.3.0 // indirect
+	github.com/moby/sys/userns v0.1.0 // indirect
+)
```
Comment (AkihiroSuda) [inline go.mod:73]: Should be merged to the `require()` section below

**Comment by apostasie** [comment]
> For test that pass i see errors:
> is this expected?

That is because the tests and test suite are retried multiple times when they fail.
Failures are still listed, but they do not fail the run if they passed at least once :/.
#3400

File: go.mod (hunk 2 above)
Comment (Shubhranshu153) [reply go.mod]: updated

**Comment by Shubhranshu153** [comment]
Updated the golint version, it seems there are some undefined pkg, which doesnt seem to be correct,

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -46,7 +46,7 @@ jobs:
           check-latest: true
           cache: true
       - name: golangci-lint
-        uses: golangci/golangci-lint-action@v3.7.0
+        uses: golangci/golangci-lint-action@v6.1.0
         with:
           version: v1.55.2
```
Comment (AkihiroSuda) [inline .github/workflows/test.yml:51]: This has to be updated
https://github.com/golangci/golangci-lint/releases/tag/v1.60.3

Comment (Shubhranshu153) [reply .github/workflows/test.yml]: updated, thanks

**Comment by apostasie** [comment]
> Updated the golint version, it seems there are some undefined pkg, which doesnt seem to be correct,

Updated golang & golinter will require additional fixes. Some were fixed here: https://github.com/containerd/nerdctl/pull/3321/files

**Comment by apostasie** [comment]
@Shubhranshu153 hard work porting these... 👍

Went through the changes - looks good - and CI is green.

Curious why the docker-compat testing suite got held back on ubuntu 22.04 - but then I don't think it matters.

**Comment by Shubhranshu153** [comment]
Not sure, in the mainline it was 22.04 so i kept 22.04.

**Comment by Shubhranshu153** [comment]
Considering the changes are ok, was wondering what are the next steps required for releasing it.
Thanks.

**Comment by apostasie** [comment]
> Not sure, in the main it was 22.04 so i kept 22.04.

Isn't it 24.04 on main? https://github.com/containerd/nerdctl/blob/main/.github/workflows/test.yml#L258-L259 (again, not a big deal anyhow, mere curiosity)

> Considering the changes are ok, was wondering what are the next steps required for releasing it.

@AkihiroSuda (or other maintainers) for this ^.

**Comment by Shubhranshu153** [comment]
ah i see i was porting some commits and it had the comment and didnt notice it has been fixed, updating it

**Comment by Shubhranshu153** [comment]
@apostasie  Not updating the docker compat one to 24.04, seems stable release doesn't have a docker 24 release
https://download.docker.com/linux/ubuntu/dists/noble/pool/stable/amd64/

File: .github/workflows/test.yml [hunk 4]
```diff
@@ -74,16 +74,20 @@ jobs:
...
       fail-fast: false
       matrix:
-        # ubuntu-20.04: cgroup v1, ubuntu-22.04: cgroup v2
+        # ubuntu-20.04: cgroup v1, ubuntu-22.04 and later: cgroup v2
         include:
           - ubuntu: 20.04
             containerd: v1.6.31
           - ubuntu: 20.04
-            containerd: v1.7.16
+            containerd: v1.7.21
```
Comment (AkihiroSuda) [inline .github/workflows/test.yml:82]: Now 1.7.22

Comment (Shubhranshu153) [reply .github/workflows/test.yml]: updated.

**Review by AkihiroSuda — APPROVED** [review]
Thanks

**Comment by AkihiroSuda** [comment]
Released
https://github.com/containerd/nerdctl/releases/tag/v1.7.7

**Pull request merged**